src/Microsoft.OData.ModelBuilder/Containers/EntitySetConfigurationOfTEntityType.cs

```diff
@@ -26,5 +27,15 @@ internal EntitySetConfiguration EntitySet
         {
             get { return (EntitySetConfiguration)Configuration; }
         }
+
+        internal DeleteRestrictionsBuilder HasDeleteRestrictions()
```

Comment:
'HasDeleteRestrictions' sounds like it should return a boolean. It looks like we're setting/getting delete restrictions

Reply:
Following the fluent pattern as is done with entity framework, you have methods HasX with returns for you a builder/configuration for which you can further configure/build the X that is there eg [here](https://docs.microsoft.com/en-us/ef/core/modeling/relationships?tabs=fluent-api%2Cfluent-api-simple-key%2Csimple-key#manual-configuration)

Reply:
~Is there any reason to adopt the EF approach for fluent patterns instead of just using 
`AddPermissions`, `AddDescription`, `AddRestrictions`?~
I was going to comment about this but noticed that the model builder does in fact already adopt a similar fluent syntax (even though it's annoyingly inconsistent)
https://docs.microsoft.com/en-us/odata/webapi/model-builder-nonconvention#open-entity-type